doc/rtd/explanation/hardening.rst

```diff
@@ -0,0 +1,18 @@
+Security Hardening
+******************
+
+Cloud-init's use case is automating cloud instance initialization, with support
+across distributions and platforms. There is a myriad of ways to harden this
+space.
+
+Follow the security hardening guidelines provided by the OSes and cloud
+platforms that your cloud-init configuration is targeting.
+
+No plain text passwords
+=======================
+
+While creating users with the
```

Comment:
Let's provide a preferred alternative to plain_text_passwds. (and passwords in general).

Do we want to tell folks to instead use `hashed_passwd` and appropriate rounds of salt?

Shall we instead suggest that folks use SSH public keys  via `ssh_import_id` or  `ssh_authorized_keys` instead and point ot security-related docs in Ubuntu for that suggestion?
https://documentation.ubuntu.com/server/how-to/security/openssh-server/#ssh-keys